pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -190,7 +191,7 @@ def restore_loops(self) -> None:
         self.trainer.fit_loop.current_epoch = self._loaded_checkpoint["epoch"]
 
         # crash if max_epochs is lower then the current epoch from the checkpoint
-        if self.trainer.max_epochs is not None and self.trainer.current_epoch > self.trainer.max_epochs:
+        if FitLoop._is_max_limit_enabled(self.max_epochs) and self.trainer.current_epoch > self.trainer.max_epochs:
```

Comment:
```suggestion
        if FitLoop._is_max_limit_enabled(self.trainer.max_epochs) and self.trainer.current_epoch > self.trainer.max_epochs:
```

check the tests
failing here